.github/workflows/build-electron.yml

```diff
@@ -39,6 +39,55 @@ jobs:
     - name: Run the build script
       run: npm run build
 
+    # Steps that use github secrets do not run in pull requests.
+    # These include codesigning & GCS deploy-related things.
+    - name: Set up Python for gsutil
+      # gsutil requires a python, which is not included on Windows
+      if: github.event_name != 'pull_request' && matrix.os == 'windows-latest'
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    - name: Set up GCP
+      # Secrets not available in PR so don't use GCP.
+      if: github.event_name != 'pull_request'
+      uses: google-github-actions/setup-gcloud@v0.2.0
+      with:
+          version: '281.0.0'
+          service_account_key: ${{ secrets.GOOGLE_SERVICE_ACC_KEY }}
+
+    - name: Code-signing setup for macOS
+      if: github.event_name != 'pull_request' && matrix.os == 'macos-latest'
+      shell: bash
+      env:
+        P12_FILE_PATH: ~/Downloads/stanford_cert.p12
+        KEYCHAIN_NAME: codesign_keychain
+        KEYCHAIN_PASS: ${{ secrets.MAC_KEYCHAIN_PASS }}
+        CERT_KEY_PASS: ${{ secrets.STANFORD_CERT_KEY_PASS }}
+
+      run: |
+        gsutil cp gs://stanford_cert/Stanford-natcap-code-signing-cert-expires-2024-01-26.p12 $P12_FILE_PATH 
+
+        bash ./scripts/setup_macos_keychain.sh 
+
+        # these env variables tell electron-builder to do code signing
+        echo "CSC_KEYCHAIN=$KEYCHAIN_NAME" >> $GITHUB_ENV
+        echo "CSC_NAME='Stanford University'" >> $GITHUB_ENV
+        echo "CSC_KEY_PASSWORD=${{ secrets.STANFORD_CERT_KEY_PASS }}" >> $GITHUB_ENV
+
+    - name: Code-signing setup for Windows
+      if: github.event_name != 'pull_request' && matrix.os == 'windows-latest'
+      env:
+        P12_FILE: Stanford-natcap-code-signing-cert-expires-2024-01-26.p12
+        CLOUDSDK_PYTHON: ${{env.pythonLocation}}\python.exe
+      run: |
+        gsutil cp gs://stanford_cert/$P12_FILE ~/$P12_FILE
+
+        # all these variables are used by electron-builder
+        echo "CSC_LINK=~/$P12_FILE" >> $GITHUB_ENV
```

Comment:
I'd forgotten what the difference is between `env:` and `$GITHUB_ENV`. Variables in `env:` are just available in this step, and variables in the `$GITHUB_ENV` file are available to future steps. So this looks just right!